crates/miden-objects/src/account/component/storage/init_storage_data.rs

```diff
@@ -0,0 +1,79 @@
+use alloc::collections::BTreeMap;
+use alloc::string::String;
+use alloc::vec::Vec;
+
+use super::StorageValueName;
+
+/// A raw word value provided via [`InitStorageData`].
+///
+/// This is used for map entries, where keys and values are supplied as either a scalar string
+/// (e.g. `"0x1234"`, `"16"`, `"BTC"`) or an array of 4 scalar elements.
+#[derive(Clone, Debug, PartialEq, Eq)]
+#[cfg_attr(feature = "std", derive(serde::Deserialize, serde::Serialize))]
+#[cfg_attr(feature = "std", serde(untagged))]
+pub enum WordValue {
+    Scalar(String),
+    Elements([String; 4]),
+}
```

Comment:
nit: I wonder if `Scalar` is the best name here. Maybe something like `Atomic` or `Unified` or `Combined` would be better?